src/common/event.cpp

```diff
@@ -80,10 +78,15 @@ std::string mir::event_type_to_string(MirEventType t)
     case mir_event_type_window_output:
         return "mir_event_type_window_output";
     default:
-        abort();
+        std::unreachable();
```

Comment:
Should we do our typical
```
#ifndef CLANG
default:
    <things>
#endif
```
dance here to make hitting this `std::unreachable()` a compile error on clang?

Reply:
MISRA doesn't like that...
```
Rule:	COV_MISRA_CPP_2008_Rule_16_2_1
Level:	4
Category:	Preprocessor Directives
Synopsis:	The pre-processor shall only be used for file inclusion and include guards.
Ruleset:	C++ Coding Standard
Tool:	Coverity 2023.12.0
```